File: app/src/main/java/ostmodern/skylark/di/AppComponent.java
package ostmodern.skylark.di;

import android.app.Application;

import javax.inject.Singleton;

import dagger.BindsInstance;
import dagger.Component;
import dagger.android.support.AndroidSupportInjectionModule;
import ostmodern.skylark.SkylarkApp;


@Singleton
@Component(modules = {
        AndroidSupportInjectionModule.class,
        AppModule.class,
        ActivityBuilderModule.class

})
public interface AppComponent {

    void inject(SkylarkApp skylarkApp);

    @Component.Builder
    interface Builder {
        @BindsInstance
        Builder application(Application application);

        AppComponent build();

    }

}

File: app/src/main/java/ostmodern/skylark/di/AppComponent.kt
package ostmodern.skylark.di

import android.app.Application

import javax.inject.Singleton

import dagger.BindsInstance
import dagger.Component
import dagger.android.support.AndroidSupportInjectionModule
import ostmodern.skylark.SkylarkApp


@Singleton
@Component(modules = [(AndroidSupportInjectionModule::class), (AppModule::class), (ActivityBuilderModule::class)])
interface AppComponent {

    fun inject(skylarkApp: SkylarkApp)

    @Component.Builder
    interface Builder {
        @BindsInstance
        fun application(application: Application): Builder

        fun build(): AppComponent

    }

}
